tests/test_client.py

```diff
@@ -0,0 +1,159 @@
+                                     'NONE',
+                                     'OFFER_CLAIMS',
+                                     'PAGE_LIKES',
+                                     'POST_ENGAGEMENT',
+                                     'THRUPLAY',
+                                     'PURCHASE',
+                                     'LISTING_INTERACTION']
+
+        self.adset_optimization_goals = ['NONE',
+                                         'APP_INSTALLS',
+                                         'AD_RECALL_LIFT',
+                                         'ENGAGED_USERS',
+                                         'EVENT_RESPONSES',
+                                         'IMPRESSIONS',
+                                         'LEAD_GENERATION',
+                                         'QUALITY_LEAD',
+                                         'LINK_CLICKS',
+                                         'OFFSITE_CONVERSIONS',
+                                         'PAGE_LIKES',
+                                         'POST_ENGAGEMENT',
+                                         'QUALITY_CALL',
+                                         'REACH',
+                                         'LANDING_PAGE_VIEWS',
+                                         'VISIT_INSTAGRAM_PROFILE',
+                                         'VALUE',
+                                         'THRUPLAY',
+                                         'DERIVED_EVENTS',
+                                         'APP_INSTALLS_AND_OFFSITE_CONVERSIONS',
+                                         'CONVERSATIONS',
+                                         'IN_APP_VALUE',
+                                         'MESSAGING_PURCHASE_CONVERSION',
+                                         'MESSAGING_APPOINTMENT_CONVERSION',
+                                         'SUBSCRIBERS',
+                                         'REMINDERS_SET']
+
+        # valid and verified objectives listed below, other objectives should be re-mapped to these
+        self.campaign_objectives = ['OUTCOME_APP_PROMOTION',
+                                    'OUTCOME_AWARENESS',
+                                    'OUTCOME_ENGAGEMENT',
+                                    'OUTCOME_LEADS',
+                                    'OUTCOME_SALES',
+                                    'OUTCOME_TRAFFIC']
+
+    def get_account_objects(self, stream):
+        assert stream in  self.stream_endpoint_map.keys(), \
+            f'Endpoint undefined for specified stream: {stream}'
+        endpoint = self.stream_endpoint_map[stream]
+        url = self.account_url + endpoint
+        params = {'access_token': self.access_token,
+                  'limit': 100}
+        LOGGER.info(f"Getting url: {url}")
+        response = requests.get(url, params)
+        response.raise_for_status()
+        LOGGER.info(f"Returning get response: {response}")
+        return response.json()
```

Comment:
It looks like this will only make one request and not continue until we either get to the end of the records or (have enough to make sure we don't need more data).  Should this be in a loop until we have a certain number of records or until there are no more records?

Also, I don't see where we set a start date.  what if all the records are ones that we wouldn't get in the sync?  I'm not sure how the API works but we should make sure we are getting the correct set of data and continue to paginate correctly ourselves.  Do we need to specify a sort order for example?

Reply:
Good point.  Had not considered how limit, start_date, and order could change things.  In general the idea was that only one request for 'limit' items would be needed and we could inspect the response to see if it had 'limit' records and a next page of data to know if there was enough there to paginate.  But this assumes the limit is always 100 and the start date is always whatever the API default is.  Once the start date differs from this then order will also matter.  Will look into passing 'limit' and start date to the request and see if there is a way to dictate order.

Reply:
Updated to pass limit and date range to the get query.